miniapp/miniapp_reduction_to_band.cpp

```diff
@@ -0,0 +1,200 @@
+
+    double gigaflops = std::numeric_limits<T>::quiet_NaN();
+    {
+      double n = matrix.size().rows();
+      double b = matrix.blockSize().rows();
+      auto add_mul = 2. / 3. * n * n * n - n * n * b;
+      gigaflops = dlaf::total_ops<T>(add_mul, add_mul) / elapsed_time / 1e9;
+    }
+
+    // print benchmark results
+    if (0 == world.rank() && run_index >= 0)
+      std::cout << "[" << run_index << "]"
+                << " " << elapsed_time << "s"
+                << " " << gigaflops << "GFlop/s"
+                << " " << matrix.size() << " " << matrix.blockSize() << " " << comm_grid.size() << " "
+                << hpx::get_os_thread_count() << std::endl;
+
+    // TODO (optional) run test
+  }
+
+  dlaf::finalize();
+  return hpx::finalize();
+}
+
+int main(int argc, char** argv) {
+  using dlaf::SizeType;
+
+  dlaf::comm::mpi_init mpi_initter(argc, argv, dlaf::comm::mpi_thread_level::multiple);
+
+  // options
+  using namespace hpx::program_options;
+  options_description desc_commandline("Usage: " HPX_APPLICATION_STRING " [options]");
+
+  // clang-format off
+  desc_commandline.add_options()
+    ("matrix-size",  value<SizeType>()   ->default_value(4),      "Matrix rows")
+    ("block-size",   value<SizeType>()   ->default_value(2),      "Block cyclic distribution size")
+    ("grid-rows",    value<int>()        ->default_value(1),      "Number of row processes in the 2D communicator")
+    ("grid-cols",    value<int>()        ->default_value(1),      "Number of column processes in the 2D communicator")
+    ("nruns",        value<int64_t>()    ->default_value(1),      "Number of runs to compute the cholesky")
+    ("nwarmups",     value<int64_t>()    ->default_value(1),      "Number of warmup runs")
+    ("check-result", value<std::string>()->default_value("none"), "Enable result checking ('none', 'all', 'last')");
+  // clang-format on
+
+  desc_commandline.add(dlaf::getOptionsDescription());
+
+  hpx::init_params p;
+  p.desc_cmdline = desc_commandline;
+  p.rp_callback = dlaf::initResourcePartitionerHandler;
+  return hpx::init(miniapp, argc, argv, p);
+}
+
+namespace {
+
+options_t parse_options(hpx::program_options::variables_map& vm) {
+  options_t opts = {
+      vm["matrix-size"].as<SizeType>(),
+      vm["block-size"].as<SizeType>(),
+      vm["grid-rows"].as<int>(),
+      vm["grid-cols"].as<int>(),
+      vm["nruns"].as<int64_t>(),
+      vm["nwarmups"].as<int64_t>(),
+      parse_red2band_check(vm["check-result"].as<std::string>()),
+  };
+
+  DLAF_ASSERT(opts.m > 0, opts.m);
+  DLAF_ASSERT(opts.mb > 0, opts.mb);
+  DLAF_ASSERT(opts.grid_rows > 0, opts.grid_rows);
+  DLAF_ASSERT(opts.grid_cols > 0, opts.grid_cols);
+  DLAF_ASSERT(opts.nruns > 0, opts.nruns);
+  DLAF_ASSERT(opts.nwarmups >= 0, opts.nwarmups);
+
+  if (opts.do_check != Red2BandCheckIterFreq::None) {
+    std::cerr << "Warning! At the moment result checking is not implemented." << std::endl;
+    opts.do_check = Red2BandCheckIterFreq::None;
+  }
```

Comment:
```suggestion
DLAF_ASSERT(opts.do_check == Red2BandCheckIterFreq::None, "Error! At the moment result checking is not implemented. Please rerun with check-result=none.");
```
I would prefer this approach as the user might not see the warning message.

Reply:
Ok, I'll do that. I just adapted the code from `miniapp_cholesky`.
Do you prefer `DLAF_ASSERT` or `DLAF_UNIMPLEMENTED`?

Reply:
I would keep the DLAF_UNIMPLEMENTED for library features.